### PR TITLE
feat(Blocks): Add conditional logic block

### PR DESCRIPTION
**Pull request by Torantulino**
Adds a basic "If statement" block, to support conditional logic in the system.

image
> Note: The red `true` text is edited in due to the GUI not showing this value due to this [known issue](https://github.com/Significant-Gravitas/AutoGPT/issues/7410).

### CODE REVIEW

File: rnd/autogpt_server/autogpt_server/blocks/if_block.py [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+            ComparisonOperator.LESS_THAN_OR_EQUAL: lambda a, b: a <= b,
+        }
+
+        try:
+            result = comparison_funcs[operator](value1, value2)
+
+            yield "result", result
+
+            if result:
+                yield "true_output", true_value
```
Comment (majdyz) [inline rnd/autogpt_server/autogpt_server/blocks/if_block.py]: the rest of the block uses `positive` `negative` should we make it consistent? we can rename the others too if it's needed.

Comment (Torantulino) [reply rnd/autogpt_server/autogpt_server/blocks/if_block.py]: Interesting question! 

I think that `Positive` and `Negative` are too ambiguous in this case, as not all conditions map clearly to a positive or negative outcome.

However I like the point you're making. I personally think perhaps `Yes` and `No` are probably the most intuitive to a non technical user.

**Comment by ntindle** [comment]
@majdyz its complaining about only two outputs here on the block. Any ideas why?

**Comment by majdyz** [comment]
The test output produced 3 values, is that the right expectation?
I suspect if block only tells you one of the value, either true or false.

**Comment by ntindle** [comment]
My expectation is it runs one of two pathways not both. The three are result, true output and false output. Do you think marking the two optional is the appropriate fix?

**Comment by majdyz** [comment]
test_input and test_output are the unittest test cases. The test case should be correct.

**Comment by ntindle** [comment]
> test_input and test_output are the unittest test cases not possible values. The test case should be correct.
> 
> The issue is simply the test case input and output doesn't match.

Welp its bedtime for me

File: rnd/autogpt_server/autogpt_server/blocks/if_block.py (hunk 1 above)
Comment (ntindle) [reply rnd/autogpt_server/autogpt_server/blocks/if_block.py]: Do we have a consensus on this one?

Comment (majdyz) [reply rnd/autogpt_server/autogpt_server/blocks/if_block.py]: yes / no sounds great!
We can change the others to yes / no too or I can follow-up on a separate PR.

**Review by ntindle — APPROVED** [review]

**Pull request merged**